Unofficial/Python/remote-control-client/aws-based-rc/client/client.py

```diff
@@ -0,0 +1,62 @@
+# Written for Python 3.6.x
+
+import json
+import os
+import random
+import string
+import subprocess
+import time
+from urllib import request
+
+from winapi import WinApi
+
+class Client:
+
+    def __init__(self, api_base, api_key):
+        self.api_base = api_base
+        self.api_key = api_key
+
+        self.client_id = self.generate_id()
+        self.current_path = os.path.dirname(os.path.realpath(__file__))
+
+    def generate_id(self):
+        client_id = ''.join(random.choices(string.ascii_letters + string.digits, k=random.randint(12, 14)))
+
+        WinApi.show_message('Client ID', client_id, WinApi.BoxType.MB_OK, WinApi.IconType.INFO)
+
+        return client_id
+
+    def get_response(self, client_id):
+        with request.urlopen(f'{self.api_base}{client_id}?key={self.api_key}') as response:
```

Comment:
Hi,

I know this is a refactoring PR and the issues exist in the original code, but should this and the request below be wrapped in a try/except URLError? or is the client just restarted if it crashes? Also below you are installing an opener on every call but it looks like that installs the opener globally so could that be moved outside the loop?

PS. I don't have write access but saw the PR while reading the code and thought I'd comment. I hope you don't mind.

Reply:
Heyo, yes they should be. Technically, since the client will be calling this from just a command line and it's not automated they can just restart it BUT I do agree that it's better to have the try/except there.

As for the opener, _probably_. I've barely used urllib and have always just stuck around with the requests library.

I'll push up a commit real quick to address these few things. If you notice anything else, feel free to comment. This was written in like 10 minutes, maybe less so there's bound to be errors.

(also, I don't mind at all :P)